conversant/demo/ui.py

```diff
@@ -16,40 +16,71 @@
 from conversant.demo import utils
 
 
+def render_bot_partial_reply(utterance, idx):
+    """Renders a partial reply message from the bot.
+
+    Args:
+        turn (str): The utterance to be rendered.
+        idx (int): The index of the turn.
+    """
+    st_message(
+        value=utterance.rstrip(),
+        animate_from=""
+        if "prev_partial_chunk" not in st.session_state
+        else st.session_state.prev_partial_chunk.lstrip(),
+        use_typewriter=True,
+        key=f"{idx}_bot",
+        avatar_style=st.session_state.bot_avatar,
+        generation_complete=(not st.session_state.partial_reply_in_progress),
+    )
```

Comment:
I can't find any docs for `st_message` -- how did you know how to use it?

Reply:
That's because I've been maintaining it 😅 - see [history here](https://github.com/yichern/st-chat/commits/main). This has been the case since https://github.com/cohere-ai/sandbox-conversant-lib/pull/8